share/rocm/cmake/ROCMChecks.cmake

```diff
@@ -26,7 +29,7 @@ function(rocm_check_toolchain_var var access value list_file)
         elseif("${base}" MATCHES ".*/CMakeFiles/${CMAKE_VERSION}$")
             set(cmake_module On)
         endif()
-        if(NOT cmake_module)
+        if(NOT cmake_module AND ROCM_CHECK_TOOLCHAIN_VAR)
             message(
```

Comment:
The short circuit should go earlier on the access modified test as no need to do the work detecting if cmake itself making the modification if we aren't messaging.   So:
if (access STREQUAL "MODIFIED_ACCESS" AND ROCM_CHECK_TOOLCHAIN_VAR)